File: src/checkout/models/order.go
package models

type PlaceOrderRequest struct {
	UserId     string     `json:"user_id"`
	Email      string     `json:"email"`
	Address    Address    `json:"address"`
	CreditCard CreditCard `json:"credit_card"`
}

type Address struct {
	StreetAddress string `json:"street_address"`
	City          string `json:"city"`
	State         string `json:"state"`
	PostalCode    string `json:"postal_code"`
	Country       string `json:"country"`
}

type CreditCard struct {
	CardNumber          string `json:"card_number"`
	CardCvv             int32  `json:"card_cvv"`
	CardExpirationMonth int32  `json:"card_expiration_month"`
	CardExpirationYear  int32  `json:"card_expiration_year"`
}

File: src/frontend/types/order.ts
export type Order = {
    user_id: string;
    email: string;
    address: Address;
    credit_card: CreditCard;
}

export type Address = {
    street_address: string;
    city: string;
    state: string;
    postal_code: string;
    country: string;
}

export type CreditCard = {
    card_number: string;
    card_cvv: number;
    card_expiration_month: number;
    card_expiration_year: number;
    card_owner: string;
}
